### PR TITLE
Fix if?/then?/else? types and add tests

### PR DESCRIPTION
**Pull request by drtyh2o**
@ostrowr I took a stab at resolving the issues I was having with #44. Feel free to make any necessary adjustments.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -3016,7 +3016,8 @@
       "version": "1.3.0",
       "resolved": "https://registry.npmjs.org/growly/-/growly-1.3.0.tgz",
       "integrity": "sha1-8QdIy+dq+WS3yWyTxrzCivEgwIE=",
-      "dev": true
+      "dev": true,
+      "optional": true
```
Comment (drtyh2o) [inline package-lock.json:3020]: These changes to `package-lock.json` were the result of my running `npm install` after cloning the repo.

**Comment by ostrowr** [comment]
Oh! Thanks so much for taking care of this!

**Review by ostrowr — CHANGES_REQUESTED** [review]
Thanks so much! Do you mind adding a test for just if/then with no /else? (I'm happy to take it over if you've had enough)

Otherwise, this looks good. Thanks for tracking this down; I imagine it must have been a bit of a slog...

**Comment by drtyh2o** [comment]
As long as you’re ok with the fundamental change to the underlying types I proposed, I can add a test for if/then. It should be just a subset of the if/then/else tests I added, although I might try testing an array of if/then (no else) schemas within an allOf for fun. :-)

**Comment by ostrowr** [comment]
> As long as you’re ok with the fundamental change to the underlying types I proposed, I can add a test for if/then. It should be just a subset of the if/then/else tests I added, although I might try testing an array of if/then (no else) schemas within an allOf for fun. :-)

Yes, definitely. It looks like the old types were just a bug. Changes to package-lock are OK too.

**Comment by drtyh2o** [comment]
@ostrowr I think this is ready now. Will you have time to review, merge, and publish a new version?

**Review by ostrowr — APPROVED** [review]
lgtm! I'll publish shortly.

**Pull request merged**